src/circuit/CircuitDefinition.js

```diff
@@ -153,7 +153,7 @@ class CircuitDefinition {
         if (newStateIndex !== undefined) {
             newVal = newStateIndex;
         }
-        if (newVal === undefined) {
+        if (newVal === undefined || newVal === 0) {
```

Comment:
You should add a unit test exercising this new code path in `CircuitDefinition.test.js`

Reply:
I couldn't find a test that uses this method, so I added one.